emailCrawler.java

```diff
@@ -31,14 +24,13 @@
 
 	public static void main(String args[]) throws IOException {
 		String url = "http://" + args[0];
-		// this list is used to store all of urls which can be discovered in
-		// home page. Actually, I can write this program
-		// by using DFS so that I can find more discoverable pages on discovered
-		// pages. But I don't do it for that it may go into a infinite loop.
+		// this list is used to store all of urls which can be discovered in home page. Actually, I can write this 
+		// program by using DFS so that I can find more discoverable pages on pages which we have discovered. 
+		// But here, I don't do it for that it may go into a infinite loop.
```

Comment:
In my original thought, I think when we found a page, there are some new links on it. And on these new links, we may found more links. So I think this will go to a infinite loop. 

To avoid infinite loop, firstly, we can use a HashSet to avoid duplicate links, which is mentioned in following note. Secondly, we may assume that we just want to search links about "Jana.com", so when we dealt with new links we found, we can check whether these new links contains "jana.com", such as:
if(link.toString.contains("jana.com")).

This is my thought about avoid infinite loop.